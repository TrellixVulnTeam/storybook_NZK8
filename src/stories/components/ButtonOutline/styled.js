// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import Button from 'components/Button';

/**
 * Determine the button's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'blue':
      return colors.iceBlue;
    default:
      return colors.transparent;
  }
}

/**
 * Determine the button's font color based on the "theme" prop
 * @param {string} theme
 * @return {string}
 */
function getFontColor(theme) {
  switch (theme) {
    case 'blue':
      return colors.brightBlue;
    default:
      return colors.slateGrey;
  }
}

/**
 * Determine the button's border color based on the "theme" prop
 * @param {string} theme
 * @return {string}
 */
function getBorderColor(theme) {
  switch (theme) {
    case 'blue':
      return colors.brightBlue;
    default:
      return '';
  }
}

export const Wrapper = styled(Button)`
  background: ${colors.transparent};
  border: 1px solid ${p => getBorderColor(p.theme)};
  color: ${p => getFontColor(p.theme)};
  min-width: inherit;
  height: auto;

  &:hover,
  &:active,
  &:visited {
    background: ${p => getAccentColor(p.theme)};
    color: ${p => getFontColor(p.theme)};
  }
`;
