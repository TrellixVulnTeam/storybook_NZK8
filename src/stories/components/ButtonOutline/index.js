// @flow

import React from 'react';
import { Wrapper } from './styled';

type Props = {
  children: React$Node,
  theme?: 'blue',
  pill?: boolean,
};

export default (props: Props) => <Wrapper {...props}>{props.children}</Wrapper>;
