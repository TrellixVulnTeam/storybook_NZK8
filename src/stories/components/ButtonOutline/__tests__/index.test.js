import React from 'react';
import { mount } from 'enzyme';
import ButtonOutline from 'components/ButtonOutline/index';

describe('<ButtonOutline />', () => {
  const Child = () => <p>Child</p>;

  it('should render correctly without props', () => {
    const sut = mount(
      <ButtonOutline>
        <Child />
      </ButtonOutline>,
    );

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with full props', () => {
    const props = {
      theme: 'blue',
      pill: true,
    };
    const sut = mount(
      <ButtonOutline {...props}>
        <Child />
      </ButtonOutline>,
    );

    expect(sut).toMatchSnapshot();
  });
});
