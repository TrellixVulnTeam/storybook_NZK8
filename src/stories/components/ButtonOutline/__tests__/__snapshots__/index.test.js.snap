// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonOutline /> should render correctly with full props 1`] = `
<_default
  pill={true}
  theme="blue"
>
  <styled__Wrapper
    pill={true}
    theme="blue"
  >
    <_default
      className="styled__Wrapper-rjpgx3-0 gsTijM"
      pill={true}
      theme="blue"
    >
      <styled__Wrapper
        className="styled__Wrapper-rjpgx3-0 gsTijM"
        onClick={[Function]}
        pill={true}
        theme="blue"
        type="button"
      >
        <button
          className="styled__Wrapper-rjpgx3-0 gsTijM styled__Wrapper-sc-6m8pxr-0 dJSzjo"
          onClick={[Function]}
          type="button"
        >
          <Child>
            <p>
              Child
            </p>
          </Child>
        </button>
      </styled__Wrapper>
    </_default>
  </styled__Wrapper>
</_default>
`;

exports[`<ButtonOutline /> should render correctly without props 1`] = `
<_default>
  <styled__Wrapper>
    <_default
      className="styled__Wrapper-rjpgx3-0 zwdkb"
    >
      <styled__Wrapper
        className="styled__Wrapper-rjpgx3-0 zwdkb"
        onClick={[Function]}
        type="button"
      >
        <button
          className="styled__Wrapper-rjpgx3-0 zwdkb styled__Wrapper-sc-6m8pxr-0 hqeYZD"
          onClick={[Function]}
          type="button"
        >
          <Child>
            <p>
              Child
            </p>
          </Child>
        </button>
      </styled__Wrapper>
    </_default>
  </styled__Wrapper>
</_default>
`;
